test/jdk/java/text/Format/ChoiceFormat/PatternsTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6801704
+ * @summary Test the expected behavior for a wide range of patterns (both
+ *          correct and incorrect). This test documents the behavior of incorrect
+ *          ChoiceFormat patterns either throwing an exception, or discarding
+ *          the incorrect portion of a pattern.
+ * @run junit PatternsTest
+ */
+
+import java.text.ChoiceFormat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+public class PatternsTest {
+
+    private static final String err1 =
+            "Each interval must contain a number before a format";
+    private static final String err2 =
+            "Incorrect order of intervals, must be in ascending order";
+
+    // Check that some valid patterns do not throw an exception
+    @ParameterizedTest
+    @MethodSource
+    public void validPatternsTest(String pattern) {
+        assertDoesNotThrow( ()-> new ChoiceFormat(pattern),
+                "Valid pattern should not have thrown an exception");
+    }
```

Comment:
For those valid cases, it may be helpful to actually format some edge cases, and check the results, especially for those odd ones. Or compare the parsed arrays.

Reply:
Good point, added some code to actually compare the formatted values to the expected results as suggested.